benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/scheduler/TestTaskScheduler.java

```diff
@@ -174,48 +178,45 @@ public synchronized void handleRunningTest(String testID) {
+          handleTestState(testID);
         } catch (BenchFlowTestIDDoesNotExistException e) {
           e.printStackTrace();
           break;
         }
 
+        testState = testModelDAO.getTestState(testID);
         testRunningState = testModelDAO.getTestRunningState(testID);
-        logger.info("handleRunningTest: prevTestState == " + prevTestState);
-        logger.info("handleRunningTest: testState == " + testState);
-        logger.info("handleRunningTest: testRunningState == " + testRunningState);
-        logger.info("handleRunningTest: prevTestRunningState == " + prevTestRunningState);
-
-        // Exit as soon as final state is executed
-        if (prevTestState == testState && (testState == WAITING || testState == TERMINATED)) {
-          exit = true;
+        logger.info("handleRunningTest: (while) prevTestState == " + prevTestState);
+        logger.info("handleRunningTest: (while) testState == " + testState);
+        logger.info("handleRunningTest: (while) prevTestRunningState == " + prevTestRunningState);
+        logger.info("handleRunningTest: (while) testRunningState == " + testRunningState);
+
+        // Exit as soon as a final state has been executed
+        if (prevTestState == WAITING || prevTestState == TERMINATED) {
+          break;
         }
 
-        /*
-          Case HANDLE_EXPERIMENT_RESULT:
-          Exit while waiting for the Experiment Manager to notify about the scheduled
-          experiment to be executed. This exits before the execution of HANDLE_EXPERIMENT_RESULT
-        
-          Case TERMINATING:
-          Check depends if the system is waiting for input from the ExperimentManager or not. In
-          the inner loop for running experiment we exit, but in the outer loop (HERE) we check if
-          we expect a result from the ExperimentManager or not. If we do not expect it we run the
-          loop again so that the TERMINATED state handling gets executed, otherwise we wait for
-          the Experiment result to trigger the TERMINATED handling.
-         */
-        else if (testRunningState == TestRunningState.HANDLE_EXPERIMENT_RESULT
-            || testRunningState == TestRunningState.TERMINATING
-                && (prevTestRunningState == TestRunningState.HANDLE_EXPERIMENT_RESULT
-                    || prevTestRunningState == TestRunningState.DETERMINE_EXECUTE_VALIDATION_SET)) {
-          exit = true;
+        // Exit while waiting for the Experiment Manager to notify about the scheduled
+        // experiment to have terminated
+        if (testRunningState == TestRunningState.HANDLE_EXPERIMENT_RESULT
+            || testRunningState == TestRunningState.DETERMINE_EXECUTE_VALIDATION_SET) {
+          break;
+        }
+
+        // Exit if we are in a loop
+        // TODO - investigate if this can be removed and instead only use abort task
+        if (prevTestRunningState == testRunningState && testState != TERMINATED) {
+          break;
```

Comment:
@jeppe-style open an issue why this is here (e.g. explain also which test fails (testTaskSchedulerIT)) and linke the code using shift